### PR TITLE
Create event seat routes

### PR DESCRIPTION
**Pull request by cll1n**

### CODE REVIEW

File: src/repositories/eventseat_repository.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		table.EventSeatCategories.AllColumns,
+	).FROM(table.EventSeats.
+		LEFT_JOIN(table.Seats, table.EventSeats.SeatID.EQ(table.Seats.ID)).
+		LEFT_JOIN(table.SeatCategories, table.Seats.SeatCategoryID.EQ(table.SeatCategories.ID)).
+		LEFT_JOIN(table.EventSeatCategories, table.EventSeatCategories.EventID.EQ(table.EventSeats.EventID).AND(table.EventSeatCategories.SeatCategoryID.EQ(table.Seats.SeatCategoryID)))).
+		WHERE(table.EventSeats.EventID.EQ(utils.MysqlUuid(eventId)))
+
+	err := stmt.Query(esr.DatabaseManager.GetDatabaseConnection(), &eventSeats)
+
+	log.Println(stmt.DebugSql())
```
Comment (Hajny25) [inline src/repositories/eventseat_repository.go:40]: remove

File: src/repositories/eventseat_repository.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	if err != nil {
+		return nil, kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if len(eventSeats) == 0 {
+		return nil, kts_errors.KTS_NOT_FOUND
```
Comment (Hajny25) [inline src/repositories/eventseat_repository.go:47]: why does this work? last time I checked, empty result set was an error

File: src/controllers/eventseats_controller.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+	if kts_err != nil {
+		return nil, nil, nil, kts_err
+	}
+
+	seatRows := make(map[int32][]models.GetSeatsForSeatSelectorDTO)
+	currentUserSeats := []models.GetSeatsForSeatSelectorDTO{}
+	var blockedUntil *time.Time
+
+	for _, seat := range *seats {
```
Comment (Hajny25) [inline src/controllers/eventseats_controller.go:31]: why not a db query for this

File: src/repositories/eventseat_repository.go (hunk 1 above)
Comment (cll1n) [reply src/repositories/eventseat_repository.go:40]: did

File: src/repositories/eventseat_repository.go (hunk 2 above)
Comment (cll1n) [reply src/repositories/eventseat_repository.go:47]: https://go.dev/tour/moretypes/12

File: src/controllers/eventseats_controller.go (hunk 3 above)
Comment (cll1n) [reply src/controllers/eventseats_controller.go:31]: Thought it was more complicated, and as I have to iterate through it anyway and sort it into appropriate places. And didn't get it to work to put it into custom models, but this should work, right?

File: src/controllers/eventseats_controller.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+			RowNr:         seat.Seat.RowNr,
+			ColumnNr:      seat.Seat.ColumnNr,
+			Available:     (seat.EventSeat.BlockedUntil == nil || seat.EventSeat.BlockedUntil.Before(time.Now()) || seat.EventSeat.UserID == nil) && !seat.EventSeat.Booked,
+			BookedByOther: (seat.EventSeat.BlockedUntil != nil && (seat.EventSeat.BlockedUntil.After(time.Now()) && (seat.EventSeat.UserID != nil && seat.EventSeat.UserID != userId))) && !seat.EventSeat.Booked,
+			Category:      seat.SeatCategory.CategoryName,
+			Price:         seat.EventSeatCategory.Price,
+		}
+
+		if seat.EventSeat.UserID == userId && !seat.EventSeat.Booked {
+			blockedUntil = seat.EventSeat.BlockedUntil
```
Comment (Hajny25) [inline src/controllers/eventseats_controller.go:43]: do it once

File: src/controllers/eventseats_controller.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+			currentUserSeats = append(currentUserSeats, currentSeat)
+		}
+
+		seatRow := seatRows[currentSeat.RowNr]
+		seatRows[currentSeat.RowNr] = append(seatRow, currentSeat)
```
Comment (Hajny25) [inline src/controllers/eventseats_controller.go:48]: you suck

File: src/controllers/eventseats_controller.go (hunk 4 above)
Comment (cll1n) [reply src/controllers/eventseats_controller.go]: did

**Review by Kieberts — CHANGES_REQUESTED** [review]

File: src/controllers/eventseats_controller.go [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+		return nil, err
+	}
+
+	err = esc.EventSeatRepo.UpdateBlockedUntilTimeForUserEventSeats(eventId, userId, &blockedUntil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &blockedUntil, nil
```
[the remark this answers is not in the document]
Comment (cll1n) [reply src/controllers/eventseats_controller.go:110]: Just one seat is unblocked, potentially there are still other seasts blocked for the user, and for those i have to return the time

**Pull request merged**